modules/userId.js

```diff
@@ -0,0 +1,431 @@
+  name: 'pubCommonId',
+  decode(value) {
+    return {
+      'pubcid': value
+    }
+  },
+  getId() {
+    // If the page includes its own pubcid object, then use that instead.
+    if (typeof window['PublisherCommonId'] === 'object') {
+      return window['PublisherCommonId'].getId();
+    }
+    // Otherwise get the existing cookie or create a new id.
+    return utils.generateUUID();
+  }
+};
+
+/**
+ * @param {SubmoduleStorage} storage
+ * @param {string} value
+ * @param {number|string} expires
+ */
+export function setStoredValue(storage, value, expires) {
+  try {
+    const valueStr = (typeof value === 'object') ? JSON.stringify(value) : value;
+    const expiresStr = (new Date(Date.now() + (expires * (60 * 60 * 24 * 1000)))).toUTCString();
+
+    if (storage.type === COOKIE) {
+      utils.setCookie(storage.name, valueStr, expiresStr);
+    } else if (storage.type === LOCAL_STORAGE) {
+      localStorage.setItem(`${storage.name}_exp`, expiresStr);
+      localStorage.setItem(storage.name, encodeURIComponent(valueStr));
+    }
+  } catch (error) {
+    utils.logError(error);
+  }
+}
+
+/**
+ * @param {SubmoduleStorage} storage
+ * @returns {string}
+ */
+export function getStoredValue(storage) {
+  let storedValue;
+  try {
+    if (storage.type === COOKIE) {
+      storedValue = utils.getCookie(storage.name);
+    } else if (storage.type === LOCAL_STORAGE) {
+      const storedValueExp = localStorage.getItem(`${storage.name}_exp`);
+      // empty string means no expiration set
+      if (storedValueExp === '') {
+        storedValue = localStorage.getItem(storage.name);
+      } else if (storedValueExp) {
+        if ((new Date(storedValueExp)).getTime() - Date.now() > 0) {
+          storedValue = decodeURIComponent(localStorage.getItem(storage.name));
+        }
+      }
+    }
+    // we support storing either a string or a stringified object,
+    // so we test if the string contains an stringified object, and if so convert to an object
+    if (typeof storedValue === 'string' && storedValue.charAt(0) === '{') {
+      storedValue = JSON.parse(storedValue);
+    }
+  } catch (e) {
+    utils.logError(e);
+  }
+  return storedValue;
+}
+
+/**
+ * test if consent module is present, applies, and is valid for local storage (purpose 1)
+ * @param {Object} consentData
+ * @returns {boolean}
+ */
+export function hasGDPRConsent(consentData) {
+  if (consentData && typeof consentData.gdprApplies === 'boolean' && consentData.gdprApplies) {
+    if (!consentData.consentString) {
+      return false;
+    }
+    if (consentData.vendorData && consentData.vendorData.purposeConsents && consentData.vendorData.purposeConsents['1'] === false) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * @param {Object[]} submodules
+ * @param {function} [processCompleted] - not required, executed when all callbacks have returned responses
+ */
+export function processSubmoduleCallbacks(submodules, processCompleted) {
```

Comment:
since this module never calls this function with `processCompleted`, when would it be used?  Is there a use case when this function would be called from outside this module?   I don't see it used by tests.

Reply:
Good call, it's for future support to delay the auction if any web requests are required, I'll remove it until (if) we need it.